content/blog/three-new-tidymodels-packages/index.Rmd

```diff
@@ -0,0 +1,192 @@
+
+Let’s create an initial split of the data (which are in the modeldata package):
+
+```{r}
+#| label: split
+library(tidymodels)
+library(filtro)
+
+set.seed(1)
+yield_split <- initial_split(modeldata::chem_proc_yield)
+yield_split
+
+yield_train <- training(yield_split)
+yield_test <- testing(yield_split)
+```
+
+We’d like to estimate the strength of the relationship between these `r ncol(modeldata::chem_proc_yield)-1` predictors and the process yield. We’ll quantify that in two ways. First is the old-fashioned [Spearman rank correlation](https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient) statistic. We can estimate these values and rank them by the absolute value of the correlations. We can also measure their value using a random forest variable importance. One quality of the predictors is that their values are correlated, so there may be some value in using an _oblique_ random forest model. This creates a collection of tree-based models with splits that are linear combinations of the selected predictors. 
+
+With filtro, we use the pre-made score objects for these measures along with the `fit()` method: 
+
+```{r}
+#| label: fit-cor
+yield_rank_res <-
+  score_cor_spearman |>
+  fit(yield ~ ., data = yield_train)
+
+# The object contains the statistics:
+yield_rank_res@results |> 
+  arrange(desc(abs(score)))
+``` 
+
+It is just as simple for the random forest model
+
+```{r}
+#| label: fit-orf
+yield_rf_res <-
+  score_imp_rf_oblique |>
+  fit(yield ~ ., data = yield_train)
+
+yield_rf_res@results |> 
+  arrange(desc(abs(score)))
+```
+
+We should probably combine the scores and do a joint ranking. To combine the two sets of statistics: 
+
+```{r}
+#| label: bind
+
+class_score_list <-
+  list(
+    yield_rank_res,
+    yield_rf_res
+  ) |>
+  bind_scores()
+
+class_score_list
+```
+
+We can accomplish a joint ranking via desirability functions. Here, we set goals for each score (i.e., maximize, minimize, etc.). The algorithm rescales their values and uses a geometric mean for an overall ranking. The desirability2 package has some nice tools for this. Here's how we do it: 
+
+```{r}
+#| label: desirability
+library(desirability2)
+class_score_list |>
+  show_best_desirability_prop(
+    maximize(cor_spearman, low = 0.25, high = 1),
+    maximize(imp_rf_oblique, scale = 2)
+  ) |> 
+  arrange(desc(.d_overall)) |> 
+  select(-starts_with(".d_max_"))
+```
+
+Using the `scale = 2` options puts more weight on the random forest results. 
+
+Now that we've looked at filtro, next up is the important package (yes, this is what we named it). 
```

Comment:
If filtro powers important (my assumption per an earlier comment), then this should be mentioned here as bridge between the two sections.